src/process_impl/mod.rs

```diff
@@ -0,0 +1,185 @@
+#![allow(
+    missing_docs,
+    missing_debug_implementations,
+    rustdoc::broken_intra_doc_links,
+    unreachable_pub
+)]
+
+use std::borrow::Cow;
+use std::ffi::OsStr;
+use std::io;
+use tokio::io::AsyncReadExt;
+use tokio::process;
+
+use super::Error;
+
+pub mod builder;
+
+mod command;
+pub use command::Command;
+
+mod child;
+pub use child::RemoteChild;
+
+#[derive(Debug)]
+pub struct Session {
```

Comment:
I don't think this has to be `pub` — `pub(crate)` should suffice.

Reply:
Same with the methods and the various inner types (like `RemoteChild`).

Reply:
Done.